opal/mca/common/cuda/common_cuda.c

```diff
@@ -1797,25 +1798,25 @@ static int mca_common_cuda_is_gpu_buffer(const void *pUserBuf, opal_convertor_t
      * and set the current context to that.  It is rare that we will not
      * have a context. */
     res = cuFunc.cuCtxGetCurrent(&ctx);
-#endif /* OPAL_CUDA_GET_ATTRIBUTES */
     if (OPAL_UNLIKELY(NULL == ctx)) {
         if (CUDA_SUCCESS == res) {
-            res = cuFunc.cuPointerGetAttribute(&ctx,
+#if !OPAL_CUDA_GET_ATTRIBUTES
+            res = cuFunc.cuPointerGetAttribute(&memCtx,
```

Comment:
They are two different code paths based on OPAL_CUDA_GET_ATTRIBUTES.  With OPAL_CUDA_GET_ATTRIBUTES set, we fill in the memCtx variable with a single call that is also filling in two other pieces of information.  Without OPAL_CUDA_GET_ATTRIBUTES, we have to make a specific call for memCtx.  Also, I need the two variables because when I do have OPAL_CUDA_GET_ATTRIBUTES, I have previously saved the memCtx variable.  I agree that when we do not have OPAL_CUDA_GET_ATTRIBUTES, we do not need memCtx.  But, that is what this bug is fixing.  Making things work with OPAL_CUDA_GET_ATTRIBUTES set.

So, the general idea is that we put the memory context into memCtx and the current context into ctx. Does that make sense?